### DISCUSSION TITLE
xterm extended mouse coordinates

### DISCUSSION THREAD

**Discussion by daviewales** [saulpw/visidata#1981] [Ideas]
Would any of these xterm [extended mouse coordinate modes](https://invisible-island.net/xterm/ctlseqs/ctlseqs.html#h3-Extended-coordinates) be useful for Visidata?

I'm thinking particularly of SGR (1006) and SGR-Pixels (1016).

**Comment by saulpw** [comment]
Are they supported by `curses` somehow?

**Reply by daviewales** [reply]
Hmm... I'm thinking you might already get SGR-1006 for free on supported Terminals.
I've just tested in iTerm and Terminal.app, and I get mouse reporting past 700 columns, so SGR-1006 is definitely working.

I don't think SGR-1016 is supported by curses.

I'll have to figure out what's happening on the few terminals which don't seem to report mouse position past a certain width.

**Reply by frosencrantz** [reply]
Maybe this is an issue with your terminal description.  https://stackoverflow.com/a/50852657

**Reply by daviewales** [reply]
OK. This is interesting.
In Windows Terminal under PowerShell, I get full width mouse reporting.
In Windows Terminal in WSL Ubuntu 20.04, it stops working past a certain width.

Following these [troubleshooting steps](https://stackoverflow.com/a/50852657), it seems that extended mouse reporting is disabled.
I'll have to see if I can figure out how to enable it.

```
$ echo $TERM
xterm-256color
$ infocmp -1x | grep XM
# no result
```

See also:
https://man7.org/linux/man-pages/man5/user_caps.5.html

It appears that `xterm-256color` is not expected to support SGR 1006 extended mouse reporting.
I suspect I need a different value for `$TERM`. Possibly `ms-terminal`, but I might need a newer version than what shipped with Ubuntu 20.04?
https://github.com/microsoft/terminal/issues/8303

**EDIT**: Updating WSL to Ubuntu 22.04 fixes this!

```
$ echo $TERM
xterm-256color
$ infocmp -1x | grep XM
        XM=\E[?1006;1000%?%p1%{1}%=%th%el%;,
```
